core/src/datayoga_core/expression.py

```diff
@@ -136,17 +124,25 @@ def exec_sql(self, data: List[Dict[str, Any]], expressions: Dict[str, str]) -> L
         """
         # use a CTE to create the in memory data structure
         data_inner = flatten_data(data)
-        cte_clause = self._get_cte(data_inner)
-
+        # builds a CTE expression for fetching in memory data
         column_names = data_inner[0].keys()
+        columns_clause = ','.join(f"[column{i+1}] as `{col}`" for i, col in enumerate(column_names))
 
-        # fetch the CTE and bind the variables
+        # values in the form of (?,?), (?,?)
+        values_clause_row = f"({','.join('?'*len(column_names))})"
+        values_clause = ','.join([values_clause_row]*len(data_inner))
+
+        subselect = f"select {columns_clause} from (values {values_clause})"
+
+        # bind the variables
         data_values = [row.get(colname) for row in data_inner for colname in column_names]
 
         # expressions clause
         expressions_clause = ", ".join([f"{expression} as `{column_name}`" for column_name,expression in expressions.items()])
-        self.conn.row_factory = pysqlite3.Row
-        statement = f"{cte_clause} select {expressions_clause} from data"
+        self.conn.row_factory = sqlite3.Row
+        # we revert to not using a real CTE because of compatibility with older SQLlite versions on Centos7
```

Comment:
In this context the revert it irrelevant (one won't know the history of it), just say:
We don't use a real CTE because of compatibility with older SQLlite versions on Centos7

Reply:
done